R/api.R

```diff
@@ -192,21 +192,28 @@ target_run_metadata <- function(runner) {
     changelog <- vcapply(changelog, scalar, USE.NAMES = FALSE)
   }
 
-  databases <- names(runner$config$database)
+  server_options <- runner$config$server_options()
   instances <- NULL
-  if (length(databases) > 0) {
-    instances <- lapply(databases, function(db) {
-      instances <- names(runner$config$database[[db]]$instances)
-      instances <- instances %||% c()
-      vcapply(instances, scalar, USE.NAMES = FALSE)
-    })
-    names(instances) <- databases
+  instances_supported <- FALSE
+  if (!isTRUE(server_options$primary)) {
+    databases <- names(runner$config$database)
+    if (length(databases) > 0) {
+      instances <- lapply(databases, function(db) {
+        instances <- names(runner$config$database[[db]]$instances)
+        instances <- instances %||% c()
+        vcapply(instances, scalar, USE.NAMES = FALSE)
+      })
+      names(instances) <- databases
+    }
+    instances_supported <- any(viapply(instances, length) > 0)
   }
 
+  git_supported <- !isTRUE(server_options$master_only) && isTRUE(runner$has_git)
+
   list(
     name = scalar(runner$config$server_options()$name),
-    instances_supported = scalar(any(viapply(instances, length) > 0)),
-    git_supported = scalar(isTRUE(runner$has_git)),
+    instances_supported = scalar(instances_supported),
```

Comment:
Sorry... not an R expert. What is the purpose of `scalar()` here?

Reply:
R has no scalar types, which causes issues with serialisation to json as the R vector containing the number 1 could be `[1]` or `1` in json. We've set up serialisation to be explicit about scalars, and `scalar` is just a utility that tells the serialiser this (https://github.com/vimc/orderly.server/blob/master/R/util.R#L25-L27)

Reply:
Ah, great. I saw that function and incorrectly assumed it was for unmarshalling incoming JSON rather than preparing the response. Thanks!